tools/spectral/ipa/__tests__/eachEnumValueMustBeUpperSnakeCase.test.js

```diff
@@ -0,0 +1,151 @@
+    },
+    errors: [],
+  },
+  {
+    name: 'invalid schema - components.schemas',
+    document: {
+      components: {
+        schemas: {
+          SchemaName: {
+            properties: {
+              exampleProperty: {
+                enum: ['exampleA', 'exampleB'],
+                type: 'string',
+              },
+            },
+          },
+        },
+      },
+    },
+    errors: [
+      {
+        code: 'xgen-IPA-123-enum-values-must-be-upper-snake-case',
+        message: 'exampleA enum value must be UPPER_SNAKE_CASE.  http://go/ipa/123',
+        path: ['components', 'schemas', 'SchemaName', 'properties', 'exampleProperty', 'enum', '0'],
+        severity: DiagnosticSeverity.Warning,
+      },
+      {
+        code: 'xgen-IPA-123-enum-values-must-be-upper-snake-case',
+        message: 'exampleB enum value must be UPPER_SNAKE_CASE.  http://go/ipa/123',
+        path: ['components', 'schemas', 'SchemaName', 'properties', 'exampleProperty', 'enum', '1'],
+        severity: DiagnosticSeverity.Warning,
+      },
+    ],
+  },
+  {
+    name: 'valid schema - paths.*',
+    document: {
+      paths: {
+        '/a/{exampleId}': {
+          get: {
+            parameters: [
+              {
+                schema: {
+                  type: 'string',
+                  enum: ['EXAMPLE_A', 'EXAMPLE_B'],
+                },
+              },
+            ],
+          },
+        },
+      },
+    },
+    errors: [],
+  },
+  {
+    name: 'invalid schema with exception - paths.*',
+    document: {
+      paths: {
+        '/a/{exampleId}': {
+          get: {
+            parameters: [
+              {
+                schema: {
+                  'x-xgen-IPA-exception': {
+                    'xgen-IPA-123-enum-values-must-be-upper-snake-case': 'reason',
+                  },
+                  type: 'string',
+                  enum: ['exampleA', 'exampleB'],
+                },
+              },
+            ],
+          },
+        },
+      },
+    },
+    errors: [],
+  },
+  {
+    name: 'invalid schema - paths.*',
+    document: {
+      paths: {
+        '/a/{exampleId}': {
+          get: {
+            parameters: [
+              {
+                schema: {
+                  type: 'string',
+                  enum: ['exampleA', 'exampleB'],
+                },
+              },
+            ],
+          },
+        },
+      },
+    },
+    errors: [
+      {
+        code: 'xgen-IPA-123-enum-values-must-be-upper-snake-case',
+        message: 'exampleA enum value must be UPPER_SNAKE_CASE.  http://go/ipa/123',
+        path: ['paths', '/a/{exampleId}', 'get', 'parameters', '0', 'schema', 'enum', '0'],
+        severity: DiagnosticSeverity.Warning,
+      },
+      {
+        code: 'xgen-IPA-123-enum-values-must-be-upper-snake-case',
+        message: 'exampleB enum value must be UPPER_SNAKE_CASE.  http://go/ipa/123',
+        path: ['paths', '/a/{exampleId}', 'get', 'parameters', '0', 'schema', 'enum', '1'],
```

Comment:
Can we have something like: `['paths', '/a/{exampleId}', 'get', 'parameters', 'parameterName', 'schema', 'enum', '1']` so it's easier to see which parameter is violating? 

Related to this I wonder if currently there are any parameters without a name, and if we should lint this

Reply:
Since the `path` follows [JSONPath](https://goessner.net/articles/JsonPath/index.html) rules, it doesn't support using names instead of indices for array elements.

I ran into the same issue with enums. I tried using the enum name instead of the index, but it didn't work. I think JSONPath doesn't recognize that as valid syntax.

Reply:
Yeah I see, thanks!